src/views/scans/viewScansList.tsx

```diff
@@ -286,80 +253,73 @@ const ScansListView: React.FunctionComponent = () => {
-            downloadReport(reportId)
-              .then(res => {
-                addAlert({
-                  title: `Report "${reportId}" downloaded`,
-                  variant: 'success',
-                  id: getUniqueId()
-                });
-                helpers.downloadData(
-                  res.data,
-                  `report_id_${reportId}_${new Date()
-                    .toISOString()
-                    .replace('T', '_')
-                    .replace(/[^\d_]/g, '')}.tar.gz`,
-                  'application/gzip'
-                );
-              })
-              .catch(err => {
-                console.error(err);
-                addAlert({
-                  title: `Report "${reportId}" failed to download. ${JSON.stringify(err.response.data)}`,
-                  variant: 'danger',
-                  id: getUniqueId()
-                });
-              });
+          onDownload={(reportId: number) => {
+            downloadReport(reportId);
           }}
           onClose={() => {
             setScanSelected(undefined);
             setScanJobs(undefined);
           }}
         />
       )}
-      {!!pendingDeleteScan && (
-        <Modal
-          variant={ModalVariant.small}
-          title={t('form-dialog.confirmation', { context: 'title_delete-scan' })}
-          isOpen={!!pendingDeleteScan}
-          onClose={() => setPendingDeleteScan(undefined)}
-          actions={[
-            <Button key="confirm" variant="danger" onClick={() => onDeleteScan()}>
-              {t('form-dialog.label', { context: 'delete' })}
-            </Button>,
-            <Button key="cancel" variant="link" onClick={() => setPendingDeleteScan(undefined)}>
-              {t('form-dialog.label', { context: 'cancel' })}
-            </Button>
-          ]}
-        >
-          Are you sure you want to delete the scan &quot;
-          {pendingDeleteScan.id}&quot;
-        </Modal>
-      )}
+      <Modal
+        variant={ModalVariant.small}
+        title={t('form-dialog.confirmation', { context: 'title_delete-source' })}
+        isOpen={pendingDeleteScan !== undefined}
+        onClose={() => setPendingDeleteScan(undefined)}
+        actions={[
+          <Button
+            key="confirm"
+            variant="danger"
+            onClick={() => {
+              if (pendingDeleteScan) {
+                deleteScans(pendingDeleteScan).finally(() => {
+                  setPendingDeleteScan(undefined);
+                  onRefresh();
+                });
+              }
+            }}
+          >
+            {t('table.label', { context: 'delete' })}
+          </Button>,
+          <Button key="cancel" variant="link" onClick={() => setPendingDeleteScan(undefined)}>
+            {t('form-dialog.label', { context: 'cancel' })}
+          </Button>
+        ]}
+      >
+        {t('form-dialog.confirmation_heading', {
+          context: 'delete-source',
+          name: pendingDeleteScan?.name
+        })}
+      </Modal>
```

Comment:
future:
similar comment to before, possible separate component and the use of view context

Reply:
added TODO